File: Server/src/main/java/pt/ist/ccu2021/queued/Server/dto/UserAccountDto.java
package pt.ist.ccu2021.queued.Server.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.*;
import pt.ist.ccu2021.queued.Server.domain.UserAccount;

import java.io.Serializable;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter @Setter
@ToString
@JsonPropertyOrder({"id", "firstName", "lastName", "email", "password"})
public class UserAccountDto implements Serializable {

    @JsonProperty("id")
    private int id;

    @JsonProperty("firstName")
    private String firstName;

    @JsonProperty("lastName")
    private String lastName;

    @JsonProperty("email")
    private String email;

    @JsonProperty("password")
    private String password;

    public UserAccountDto(UserAccount user){
        id = user.getId();
        firstName = user.getFirstName();
        lastName = user.getLastName();
        email = user.getEmail();
        password = user.getPassword();
        //dateOfBirth = user.getDateOfBirth().toInstant().toEpochMilli();

    }

    public UserAccount toDomain(){
        return UserAccount.builder().firstName(firstName).lastName(lastName).email(email)
                .password(password)/*.dateOfBirth(new Date(dateOfBirth))*/.build();
    }
}

File: Queued/lib/dto/UserAccountDto.dart
class UserAccountDto {

  UserAccountDto({this.id, this.firstName, this.lastName, this.email, this.password});

  int id;

  String firstName;

  String lastName;

  String email;

  String password;


  UserAccountDto.fromJson(Map<String, dynamic> json) {
    this.id = json['id'];
    this.firstName = json['firstName'];
    this.lastName = json['lastName'];
    this.email = json['email'];
    this.password = json['password'];
  }

  Map<String, dynamic> toJson(){
      return {'id': this.id.toString(), 'firstName': this.firstName, 'lastName': this.lastName, 'email': this.email, 'password': this.password};
  }


}
